modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -801,7 +801,7 @@ public WorkerActionStateMachine get(String workerId) {
         }
         
         public boolean everyoneDone() {
-            return this.workerActionStateMachines.values().stream().map(e -> e.done()).reduce(true, (a, b) -> a && b);
+            return !this.workerActionStateMachines.values().stream().anyMatch(e -> !e.done());
```

Comment:
@hisamith . Thanks for simplifying this. But we can simplify this further using allMatch(). Then we don't need to check the negative condition. 

also e -> e.done() can be replaced with method referance (i.e WorkerActionStateMachine::done)

Reply:
@hasithaa Thanks for the great feedback and I have incooperated the requested changes